google/earthenterprise
# Copy uninstall scripts

This PR solves #303 which is to copy the uninstall scripts over to `/opt/google` so that the user can uninstall the program without having to keep around the source code.  In particular, the changes make it so that the uninstall scripts and their dependencies are moved over to the `/common` directory during the staging process, and then during the install process the installation script moves over its respective uninstall script along with the `common.sh` dependency to the `/install` directory.  The `version.txt` dependency is moved to the base `/opt/google` directory since the uninstall scripts look for that file in the superdirectory from where they are.

As part of the development process, I and @tst-lsavoie discovered a bug with `khEnvironment.py` where we didn't check to see if a builder existed before we tried to use it in the alias function.  This PR includes a fix for that bug since it was a blocker on implementing the PR.

Verification:
1) Build and stage-install OpenGEE.  All files should be in the `/common` subdirectory in the staging area.
2) Install either or both fusion and server.  `version.txt` should be in the base `/opt/google` directory, `common.sh` should be in `/install` along with the respective uninstall script for whichever install scripts you ran.
3) When you run one or both uninstall scripts, then `/opt/google` should be empty besides `version.txt` and the `/install` directory.  If both are installed and only one product is uninstalled, then `/opt/google` should only be missing whatever is taken away but leaving whatever is necessary for the other product to work.  The uninstall scripts and their dependencies are not removed in either case since that is under the scope of #488.
